src/Connect.js

```diff
@@ -1,68 +1,441 @@
+   * @param    {String}      [opts.accountType]           Ethereum account type: "general", "segregated", "keypair", "devicekey" or "none"
+   * @param    {Boolean}     [opts.isMobile]              Configured by default by detecting client, but can optionally pass boolean to indicate whether this is instantiated on a mobile client
+   * @param    {Boolean}     [opts.storage=true]          When true, object state will be written to local storage on each state cz-conventional-change
+   * @param    {Function}    [opts.transport]             Configured by default by detecting client, but can optionally pass boolean to indicate whether this is instantiated on a mobile client
+   * @param    {Function}    [opts.mobileTransport]       Configured by default by detecting client, but can optionally pass boolean to indicate whether this is instantiated on a mobile client
+   * @param    {Object}      [opts.muportConfig]          Configuration object for muport did resolver. See [muport-did-resolver](https://github.com/uport-project/muport-did-resolver)
+   * @param    {Object}      [opts.ethrConfig]            Configuration object for ethr did resolver. See [ethr-did-resolver](https://github.com/uport-project/ethr-did-resolver)
+   * @param    {Object}      [opts.registry]              Configuration for uPort DID Resolver (DEPRACATED) See [uport-did-resolver](https://github.com/uport-project/uport-did-resolver)
+   * @return   {Connect}                                  self
    */
-
   constructor (appName, opts = {}) {
-    super(appName, opts)
-    this.uriHandler = opts.uriHandler || openQr
-    this.mobileUriHandler = opts.mobileUriHandler || mobileUriHandler
-    this.closeUriHandler = opts.closeUriHandler || (this.uriHandler === openQr ? closeQr : undefined)
+    // Config
+    this.appName = appName || 'uport-connect-app'
+    this.network = network.config.network(opts.network)
+    this.provider = opts.provider || new HttpProvider(this.network.rpcUrl)
+    this.accountType = opts.accountType
+    this.isOnMobile = opts.isMobile === undefined ? isMobile() : opts.isMobile
+    this.storage = opts.storage === undefined ? true : opts.storage
+    // Transports
+    this.transport = opts.transport || connectTransport(appName)
+    this.mobileTransport = opts.mobileTransport || transport.url.send()
+    this.onloadResponse = transport.url.getResponse()
+    this.PubSub = PubSub
+    transport.url.listenResponse((err, payload) => {
+      if (payload) this.PubSub.publish(payload.id, {res: payload.res, data: payload.data})
+    })
+
+    // State
+    this.did = null
+    this.mnid = null
+    this.address = null
+    this.firstReq = true // Add firstReq?
+    this.doc = null
+
+    // Load any existing state if any
+    if (this.storage)  this.getState()
+    if (!this.keypair) this.keypair = Credentials.createIdentity()
+    if (this.storage)this.setState()
```

Comment:
Remembering to do this after setting state each time is error prone. Create a single state object as mentioned before and mutate it through a function. This can then be automatically persisted if necessary.